packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/OutputLibrary.cs

```diff
@@ -10,58 +10,60 @@ namespace Microsoft.Generator.CSharp
 {
     public class OutputLibrary
     {
-        private IReadOnlyList<EnumProvider>? _enums;
-        private IReadOnlyList<ModelProvider>? _models;
+        private IReadOnlyList<TypeProvider>? _enums;
+        private IReadOnlyList<TypeProvider>? _models;
         private IReadOnlyList<ClientProvider>? _clients;
 
         public OutputLibrary()
         {
-            EnumMappings = new Dictionary<InputEnumType, EnumProvider>();
-            ModelMappings = new Dictionary<InputModelType, ModelProvider>();
+            EnumMappings = new Dictionary<InputEnumType, TypeProvider>();
```

Comment:
Since this is only for caching to not create the same thing twice the cache should live with the class that does the creation.

Otherwise every consumer of TypeFactory would have to implement their own caching.